rust/connlib/tunnel/src/peer.rs

```diff
@@ -508,6 +502,32 @@ impl ClientOnGateway {
     pub fn id(&self) -> ClientId {
         self.id
     }
+
+    /// Check all [`TranslationState`]s for dead but used IPs.
+    ///
+    /// We don't want to be spamming this warning but it also shouldn't go unnoticed.
+    /// Thus, the strategy is to only print the log if:
+    /// - An IP has recently been used (we have seen outgoing traffic in the last 30s)
+    /// - An IP has not responded in the last 10s
+    fn check_for_dead_ips(&self, now: Instant) {
+        let mut dead_ips = BTreeMap::<DomainName, BTreeSet<IpAddr>>::new();
```

Comment:
Why use b-trees instead of hashmaps?

Reply:
I figured it is nice to ensure that these are always sorted so in case they do get emitted multiple times, it easier to glance over!